client/boot/locale.js

```diff
@@ -68,10 +77,13 @@ export const setupLocale = ( currentUser, reduxStore ) => {
 		// For logged out Calypso pages, set the locale from query param
 		const pathLocaleSlug = getLocaleFromQueryParam();
 		pathLocaleSlug && reduxStore.dispatch( setLocale( pathLocaleSlug, '' ) );
-	} else {
+	} else if ( getLocaleFromPathname() ) {
 		// For logged out Calypso pages, set the locale from slug
 		const pathLocaleSlug = getLocaleFromPathname();
 		pathLocaleSlug && reduxStore.dispatch( setLocale( pathLocaleSlug, '' ) );
+	} else {
```

Comment:
I don't know if we'd want to use the browser language to set the UI in Calypso enabled across the entire app, as it seems like it may cause a lot of unwanted side effects.

For example, the first side-effect that I've noticed was on `/themes` - it's initially server side rendered in English, then it flashes and shifts the layout after detecting the browser language. I'd assume we'd have the same behavior in other places as well.